File: java/src/main/java/cn/az/code/year2020/july/ClimbStairs.java
package cn.az.code.year2020.july;

/**
 * @author az
 * @since 07/31/20
 */
public class ClimbStairs {

    public static void main(String[] args) {
        ClimbStairs cs = new ClimbStairs();
        System.out.println(cs.climb(4));
    }

    int climb(int n) {
        if (n == 1 || n == 2) {
            return n;
        }
        int[] dp = new int[n + 1];
        dp[0] = 1;
        dp[1] = 2;
        for (int i = 2; i <= n; i++) {
            dp[i] = dp[i - 1] + dp[i - 2];
        }
        return dp[n-1];
    }
}

File: code/src/array/climb-stairs.cc
class Solution
{
public:
    int climbStairs(int n)
    {
        if (n <= 2)
        {
            return n;
        }
        int f1 = 1, f2 = 2, f3;
        int i = 3;
        while (i < n + 1)
        {
            f3 = f1 + f2;
            f1 = f2;
            f2 = f3;
        }
        return f3;
    }
};